pkg/tunneler/tunnel.go

```diff
@@ -120,27 +109,25 @@ func SyncerTunnelURL(host, ws, target string) (string, error) {
 	if err != nil || hostURL.Scheme != "https" || hostURL.Host == "" {
 		return "", fmt.Errorf("wrong url format, expected https://host<:port>/<path>: %w", err)
 	}
-	host = strings.Trim(host, "/")
-	return host + "/clusters/" + ws + "/apis/" + workloadv1alpha1.SchemeGroupVersion.String() + "/synctargets/" + target, nil
+	return url.JoinPath(hostURL.String(), "clusters", ws, "apis", workloadv1alpha1.SchemeGroupVersion.String(), "synctargets", target)
 }
 
 // WithSyncerTunnel adds an HTTP Handler that handles reverse connections and reverse proxy requests using 2 different paths exposing two synctarget subresources:
 //
-// https://host/clusters/<ws>/apis/workload.kcp.io/v1alpha1/synctargets/<name>/connect establish reverse connections and queue them so it can be consumed by the dialer
+// https://host/clusters/<ws>/apis/workload.kcp.io/v1alpha1/synctargets/<name>/tunnel establish reverse connections and queue them so it can be consumed by the dialer
 // https://host/clusters/<ws>/apis/workload.kcp.io/v1alpha1/synctargets/<name>/proxy/{path} proxies the {path} through the reverse connection identified by the cluster and syncer name
-func WithSyncerTunnel(apiHandler http.Handler) http.HandlerFunc {
-	pool := newTunnelPool()
+func (tn *Tunneler) SyncerTunnelHandler(apiHandler http.Handler) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		logger := klog.FromContext(r.Context())
 
 		// fall through, syncer tunnels URL should contain the synctarget gvr.
-		if !strings.Contains(r.RequestURI, "workload.kcp.io/v1alpha1/synctargets") {
```

Comment:
Can't we use `RequestInfo` to do those checks here now ?

Reply:
done